gcloud-java-contrib/gcloud-java-nio/src/test/java/com/google/gcloud/storage/contrib/nio/ITGcsNio.java

```diff
@@ -0,0 +1,360 @@
+    final ByteBuffer buf2a = ByteBuffer.allocate(100);
+    final ByteBuffer buf2b = ByteBuffer.allocate(100);
+    SeekableByteChannel chan = Files.newByteChannel(path, StandardOpenOption.READ);
+    assertThat(chan.size()).isEqualTo(size);
+    reallyRead(chan, buf1a);
+    long dest = size / 2;
+    chan.position(dest);
+    reallyRead(chan, buf2a);
+    // now go back and read it again
+    // (we do 2 locations because 0 is sometimes a special case).
+    chan.position(0);
+    reallyRead(chan, buf1b);
+    chan.position(dest);
+    reallyRead(chan, buf2b);
+    // if the two spots in the file have the same contents, then this isn't a good file for this
+    // test.
+    assertThat(buf1a.array()).isNotEqualTo(buf2a.array());
+    assertThat(buf1a.array()).isEqualTo(buf1b.array());
+    assertThat(buf2a.array()).isEqualTo(buf2b.array());
+  }
+
+  @Test
+  public void testCreate() throws IOException {
+    CloudStorageFileSystem testBucket = getTestBucket();
+    Path path = testBucket.getPath(PREFIX + randomSuffix());
+    try {
+      // file shouldn't exist initially. If it does it's either because it's a leftover
+      // from a previous run (so we should delete the file)
+      // or because we're misconfigured and pointing to an actually important file
+      // (so we should absolutely not delete it).
+      // So if the file's here, don't try to fix it automatically, let the user deal with it.
+      assertThat(Files.exists(path)).isFalse();
+
+      Files.createFile(path);
+      // now it does, and it has size 0.
+      assertThat(Files.exists(path)).isTrue();
+      long size = Files.size(path);
+      assertThat(size).isEqualTo(0);
+    } finally {
+      // let's not leave files around
+      Files.deleteIfExists(path);
+    }
+  }
+
+  @Test
+  public void testWrite() throws IOException {
+    CloudStorageFileSystem testBucket = getTestBucket();
+    Path path = testBucket.getPath(PREFIX + randomSuffix());
+    try {
+      // file shouldn't exist initially. If it does it's either because it's a leftover
+      // from a previous run (so we should delete the file)
+      // or because we're misconfigured and pointing to an actually important file
+      // (so we should absolutely not delete it).
+      // So if the file's here, don't try to fix it automatically, let the user deal with it.
+      assertThat(Files.exists(path)).isFalse();
+
+      Files.write(path, FILE_CONTENTS, UTF_8);
+      // now it does.
+      assertThat(Files.exists(path)).isTrue();
+
+      // let's check that the contents look OK.
+      ByteBuffer buf = ByteBuffer.allocate(100);
+      SeekableByteChannel chan = Files.newByteChannel(path, StandardOpenOption.READ);
+      reallyRead(chan, buf);
+      byte[] gotBytes = buf.array();
+      byte[] wantBytes = FILE_CONTENTS.get(0).getBytes(UTF_8);
+      for (int i = 0; i < wantBytes.length; i++) {
+        assertThat(gotBytes[i] == wantBytes[i]).isTrue();
+      }
+
```

Comment:
I think it would be better if we adhere to the same style of existing code in this repository if possible.